### ISSUE TITLE
Add a write buffer test that removes entries then resets to earliest

### ISSUE THREAD

**Issue by carols10cents** [influxdata/influxdb_iox#4651]
[From here:](https://github.com/influxdata/influxdb_iox/pull/4638#discussion_r877821439)

> This [`WriteBufferStreamHandler::reset_to_earliest`] needs a generic test here in core.rs. The most basic version would just test that a call to this jumps back to the earliest sequence number (which is likely 0).
>
> However I would like if we could actually have a test that involves deleting/forgetting data. For that we would need to extend TestContext with some "delete entries prior to X" method. We have all the functionality in place (for all WB implementations) to achieve that, it's just not wired up. If you want I can implement that. If you wanna have this PR merged fast, just implement the easy test (= reset to 0), open a follow-up ticket for the advanced test (= interaction w/ deletes) and assign that that to me.

Yeah, I don't know how to delete entries from all the write buffer implementations, so I'm giving this to you :)

**Comment by crepererum** [comment]
So we have three write buffer types:

- kafka
- file
- mock

For *file* and *mock* the implementation of "delete messages prior to this sequence number" is simple. For *kafka* we support this in `rskafka` as well. However we use redpanda instead of the official Kafka implementation in our CI and redpanda doesn't support the required API (see https://github.com/redpanda-data/redpanda/issues/2648 ). So I consider this blocked on upstream (or until we change the broker implementation in our CI pipeline).
